diffrax/_event.py

```diff
@@ -21,7 +22,32 @@ class Event(eqx.Module):
     """
 
     cond_fn: PyTree[Callable[..., Union[BoolScalarLike, RealScalarLike]]]
-    root_finder: Optional[optx.AbstractRootFinder] = None
+    bidirect: PyTree[bool]
+    root_finder: Optional[optx.AbstractRootFinder]
+
+    def __init__(
+        self,
+        cond_fn,
+        root_finder: Optional[optx.AbstractRootFinder] = None,
+        bidirect: Union[bool, PyTree[bool]] = True,
```

Comment:
Perhaps this should be a `PyTree[None | bool]`: handling the cases of upcrossing-only, downcrossing-only, or either crossing. WDYT? (No strong feelings from me, since one crossing can easily be constructed from the other.)